graphistry/umap_utils.py

```diff
@@ -31,17 +31,57 @@ def lazy_umap_import_has_dependancy():
     try:
         import warnings
         warnings.filterwarnings("ignore")
-        import umap  # noqa
+        import umap
         return True, 'ok', umap
     except ModuleNotFoundError as e:
         return False, e, None
+    
+def lazy_cuml_import_has_dependancy():
+    try:
+        import warnings
+        warnings.filterwarnings("ignore")
+        import cuml
+        return True, 'ok', cuml
+    except ModuleNotFoundError as e:
+        return False, e, None
 
 def assert_imported():
-    has_dependancy, import_exn, _ = lazy_umap_import_has_dependancy()
-    if not has_dependancy:
+    has_umap_dependancy_, import_umap_exn, _ = lazy_umap_import_has_dependancy()
+    if not has_umap_dependancy_:
         logger.error("UMAP not found, trying running "
                      "`pip install graphistry[ai]`")
-        raise import_exn
+        raise import_umap_exn
+
+def assert_imported_cuml():
+    has_cuml_dependancy_, import_cuml_exn, _ = lazy_cuml_import_has_dependancy()
+    if not has_cuml_dependancy_:
+        logger.error("cuML not found, trying running "
+                     "`pip install cuml`")
+        raise import_cuml_exn
+
+UMAPEngineConcrete = Literal["none", "cuml", "umap_learn"]
```

Comment:
`"none"` seems like an illegal value in this case -- how do we do a umap w/ no engine?

(it's valid for feature engineering b/c that means a no-op passthrough of features)